sklearn/compose/tests/test_column_transformer.py

```diff
@@ -649,25 +657,53 @@ def test_column_transformer_get_feature_names():
     ct.fit(X)
     assert_equal(ct.get_feature_names(), ['col0__a', 'col0__b', 'col1__c'])
 
-    # passthrough transformers not supported
+    # drop transformer
+    ct = ColumnTransformer(
+        [('col0', DictVectorizer(), 0), ('col1', 'drop', 1)])
+    ct.fit(X)
+    assert_equal(ct.get_feature_names(), ['col0__a', 'col0__b'])
+
+    # passthrough transformer
     ct = ColumnTransformer([('trans', 'passthrough', [0, 1])])
     ct.fit(X)
-    assert_raise_message(
-        NotImplementedError, 'get_feature_names is not yet supported',
-        ct.get_feature_names)
+    assert_equal(ct.get_feature_names(), ['trans__0', 'trans__1'])
 
     ct = ColumnTransformer([('trans', DictVectorizer(), 0)],
                            remainder='passthrough')
     ct.fit(X)
-    assert_raise_message(
-        NotImplementedError, 'get_feature_names is not yet supported',
-        ct.get_feature_names)
+    assert_equal(ct.get_feature_names(), ['trans__a', 'trans__b', 1])
```

Comment:
Not entirely happy with the mix of string and numeric types.

Reply:
I know what you mean. The alternative would be to cast the array indices to string, would that be better?

Reply:
This comes a bit in the general "get feature names" discussion domain. For arrays, we might give default names like "x0", "x1", ... but that's not fully decided yet.

Reply:
"x0", "x1", ... seems to be used in a few other transformers (e.g. PolynomialFeatures), and is mentioned in scikit-learn/enhancement_proposals#18 as well. Will this be dependent on that SLEP then?

Reply:
That SLEP could certainly inform the choice. I'd be most happy with "x0", "x1", ... here for now.

Reply:
Should the indices be the same as the indices in the input array, or should they be from 0 to the number of passthrough columns? I think that keeping the indices the same as the input would make them easier to interpret, but this goes against the behaviour of something like PolynomialFeatures, or even the behaviour implemented here for transformers with no get_feature_names, where they always start counting from "x0" .

If starting from 0, the strings would probably need to start with something like 'passthrough__' to avoid confusion.

Reply:
Number according to the input features, please.

Reply:
I've added this now, thanks